core/test/Test.re

```diff
@@ -773,16 +757,278 @@ let suites = [
+          CommitChanges,
+        ],
+        Implementation.mountLog^,
+      );
+    },
+  ),
+  (
+    "Test top level replace elements",
+    `Quick,
+    () => {
+      let root = Implementation.{name: "root", element: View};
+
+      let previousReactElement = <Components.Text key=1 title="x" />;
+      let nextReactElement = <Components.Text key=2 title="y" />;
+
+      let beforeUpdate = TestRenderer.render(previousReactElement);
+      let _ = MountLog.mountRenderedElement(root, beforeUpdate);
+
+      assertMountLog(
+        ~label="It correctly mounts top level element (for replace elemets)",
+        [
+          Implementation.BeginChanges,
+          MountChild(root, {name: "Text", element: Text("x")}, 0),
+          CommitChanges,
+        ],
+        Implementation.mountLog^,
+      );
+
+      let (afterUpdate, topLevelUpdateLog) =
+        RenderedElement.update(
+          ~previousReactElement,
+          ~renderedElement=beforeUpdate,
+          nextReactElement,
+        );
+
+      MountLog.applyTopLevelUpdate(root, afterUpdate, topLevelUpdateLog);
+      assertMountLog(
+        ~label="It correctly mounts `ReplaceElements topLevelUpdate",
+        [
+          Implementation.BeginChanges,
+          UnmountChild(root, {name: "Text", element: Text("x")}),
+          MountChild(root, {name: "Text", element: Text("y")}, 0),
+          CommitChanges,
+        ],
+        Implementation.mountLog^,
+      );
+    },
+  ),
+  (
+    "Test top level prepend",
+    `Quick,
+    () => {
+      let root = Implementation.{name: "root", element: View};
+
+      GlobalState.useTailHack := true;
+      let key1 = Key.create();
+      let key2 = Key.create();
+      let commonElement = [<Components.Text key=key1 title="x" />];
+
+      let previousReactElement = listToElement(commonElement);
+      let nextReactElement =
+        listToElement([
+          <Components.Text key=key2 title="y" />,
+          ...commonElement,
+        ]);
+
+      let beforeUpdate = TestRenderer.render(previousReactElement);
+      let _ = MountLog.mountRenderedElement(root, beforeUpdate);
+
+      assertMountLog(
+        ~label="It correctly mounts top level list (for prepend)",
+        [
+          Implementation.BeginChanges,
+          MountChild(root, {name: "Text", element: Text("x")}, 0),
+          CommitChanges,
+        ],
+        Implementation.mountLog^,
+      );
+
+      let (afterUpdate, topLevelUpdateLog) =
+        RenderedElement.update(
+          ~previousReactElement,
+          ~renderedElement=beforeUpdate,
+          nextReactElement,
+        );
+
+      MountLog.applyTopLevelUpdate(root, afterUpdate, topLevelUpdateLog);
+      assertMountLog(
+        ~label="It correctly mounts `Prepend topLevelUpdate",
+        [
+          Implementation.BeginChanges,
+          MountChild(root, {name: "Text", element: Text("y")}, 0),
+          CommitChanges,
+        ],
+        Implementation.mountLog^,
+      );
```

Comment:
We're not testing `ContentChanged`

Reply:
We are. `Test top level reorder` is the one covering `` `ContentChanged(`NoChange)``

Reply:
That's not correct. We are testing `UpdateLog.ContentChanged` but not in the context of mount log. For mount log we should check if reconfigure instance is called.